osquery/tables/system/windows/bios_info.cpp

```diff
@@ -0,0 +1,147 @@
+  } else if (std::find(vLenovo.begin(), vLenovo.end(), manufacturer) !=
+             vLenovo.end()) {
+    manufacturer = "lenovo";
+  } else if (std::find(vDell.begin(), vDell.end(), manufacturer) !=
+             vDell.end()) {
+    manufacturer = "dell";
+  }
+
+  return manufacturer;
+}
+
+QueryData genBiosInfo(QueryContext& context) {
+  QueryData results;
+  std::string manufacturer;
+
+  WmiRequest wmiComputerSystemReq(
+      "select Manufacturer from Win32_ComputerSystem");
+  std::vector<WmiResultItem>& wmiComputerSystemResults =
+      wmiComputerSystemReq.results();
+
+  if (!wmiComputerSystemResults.empty()) {
+    wmiComputerSystemResults[0].GetString("Manufacturer", manufacturer);
+    manufacturer = getManufacturer(manufacturer);
+  } else {
+    manufacturer = "N/A";
+  }
+
+  if (queryMap.find(manufacturer) != queryMap.end()) {
+    WmiRequest wmiBiosReq(std::get<0>(queryMap.at(manufacturer)),
+                          (std::get<1>(queryMap.at(manufacturer))));
+    std::vector<WmiResultItem>& wmiResults = wmiBiosReq.results();
+
+    for (unsigned int i = 0; i < wmiResults.size(); ++i) {
+      Row r;
+
+      if (manufacturer == "hp") {
+        std::string value;
+        boost::smatch matches;
+        wmiResults[i].GetString("Name", r["Name"]);
+        wmiResults[i].GetString("Value", value);
+
+        if (boost::regex_search(value, matches, kHPBiosSettingRegex)) {
+          r["Value"] = std::string(matches[1]);
+        } else {
+          r["Value"] = value;
+        }
+
+      } else if (manufacturer == "lenovo") {
+        std::string currentSetting;
+        std::vector<std::string> settings;
+        wmiResults[i].GetString("CurrentSetting", currentSetting);
+        settings = osquery::split(currentSetting, ",");
+
+        if (settings.size() != 2) {
+          continue;
+        }
+        r["Name"] = settings[0];
+        r["Value"] = settings[1];
+
+      } else if (manufacturer == "dell") {
+        std::vector<std::string> vCurrentValue;
+        std::vector<std::string> vPossibleValues;
+        std::vector<std::string> vPossibleValuesDescription;
+        wmiResults[i].GetString("AttributeName", r["Name"]);
+        wmiResults[i].GetVectorOfStrings("CurrentValue", vCurrentValue);
+        wmiResults[i].GetVectorOfStrings("PossibleValues", vPossibleValues);
+        wmiResults[i].GetVectorOfStrings("PossibleValuesDescription",
+                                         vPossibleValuesDescription);
+
+        if (vCurrentValue.size() == 1 && !vPossibleValues.empty()) {
+          auto pos = std::find(
+              vPossibleValues.begin(), vPossibleValues.end(), vCurrentValue[0]);
+          if (pos != vPossibleValues.end()) {
+            r["Value"] =
+                vPossibleValuesDescription[pos - vPossibleValues.begin()];
+          } else {
+            r["Value"] = "N/A";
+          }
+
+        } else if (vCurrentValue.size() > 1) {
+          std::ostringstream oValueConcat;
```

Comment:
Instead of all of this, make use of our `join` [helper function here](https://github.com/facebook/osquery/blob/master/osquery/core/conversions.h#L110)